core/types/transaction_signing.go

```diff
@@ -514,6 +543,117 @@ func (s MikoSigner) Payer(tx *Transaction) (common.Address, error) {
 	return payerInternal(s, tx)
 }
 
+type EnaSigner struct {
+	EIP155Signer
```

Comment:
I think making EnaSigner "inherits" MikoSigner avoids duplicated Sender, SignatureValues, Hash code, we just need to override Equal and Payer